tests/expected_tests.cc

```diff
@@ -0,0 +1,39 @@
+#include "gtest/gtest.h"
+#include "stout/expected.h"
+
+stout::expected<int, std::string> divide(int divident, int divider) {
+  if (!divider) {
+    return stout::make_unexpected("divide by zero");
+  } else {
+    return divident / divider;
+  }
+}
+
+TEST(Expected, SucceedResult) {
+  const auto result = divide(6, 3);
+
+  EXPECT_TRUE(result.has_value());
+  EXPECT_EQ(2, *result);
+}
+
+TEST(Expected, DivideByZero) {
+  const auto result = divide(6, 0);
+
+  EXPECT_TRUE(!result.has_value());
+  EXPECT_EQ("divide by zero", result.error());
+}
+
+TEST(Expected, MakeUnexpected) {
+  auto error = []() {
+    return stout::make_unexpected("error");
+  };
+  const auto result = error();
+
+  EXPECT_STREQ("error", result.value());
```

Comment:
`make_unexpected` should use `std::string` as an error type, right?

Prefer `EXPECT_EQ` for `std::string`. (I'm almost willing to advise avoiding EXPECT_STREQ entirely, since the cost of creating "unnecessary" `std::string`s in test code is low, and the safety/nicety of using standard macros and C++ types might often be worth it)

Reply:
Nope, the error type here is `const char *const &`. That's why I use `EXPECT_STREQ` instead of `EXPECT_EQ`. Otherwise this test will fail with the following message:
![log](https://user-images.githubusercontent.com/68085133/181233373-ce5fd209-fb03-45ea-922a-01b3820779e1.png)

Reply:
Per the above comment, this might be fixable if you change the lambda's `make_unexpected("error")` to `make_unexpected<std::string>("error")`.

But ooh, I thought that we were default-defining `std::string` as the error type. Since `stout::expected<Foo>` always have an error type of `std::string`, shouldn't `stout::make_unexpected` do the same?

Maybe we could also provide this overload of make_unexpected in `stout/expected.h` (dunno if this compiles):

```c++
template <typename Error = std::string>
using make_unexpected = tl::make_unexpected<Error>;
```

Reply:
Absolutely great idea! But unfortunately your alias that you've defined above will not compile because it refers to a non-type template.
![log2](https://user-images.githubusercontent.com/68085133/181260630-28c8b960-21f3-4dbb-aa68-2e06f6012759.png)

But the following bellow will work for us:

```
template <typename Error>
auto make_unexpected = tl::make_unexpected<Error>;
```

One note: we should explicitly define template argument and cannot omit it:(
For example: 
```
stout::make_unexpected<std::string>("divide by zero")
```

Reply:
Like I mean that `stout::make_unexpected("divide by zero");` will not compile if we do the following:
```
template <typename Error = std::string>
auto make_unexpected = tl::make_unexpected<Error>;
```

Reply:
> Like I mean that `stout::make_unexpected("divide by zero");` will not compile if we do the following:
> 
> ```
> template <typename Error = std::string>
> auto make_unexpected = tl::make_unexpected<Error>;
> ```

Ah, right :/ 

Ok, one more silly idea:

```c++
stout
```

Reply:
Will try!

Reply:
@CodingCanuck , well, it works for us! I think this is also not bad idea! Thanks:)
```
template <typename ErrorInputType>
unexpected<std::string> make_unexpected(ErrorInputType&& e) {
  return tl::make_unexpected<std::string>(std::move(e));
}
```